src/utils/buildCssOverridesStyle.ts

```diff
@@ -0,0 +1,17 @@
+export type CssVariableOverrides = {
+  [key: string]: string;
+};
+
+export const buildCssOverridesStyle = (
```

Comment:
Maybe we call it `applyTheme`?

I also think it should take in the document directly and we pull off the theme field, which will be part of the site entity and have a specific name. Presumably that field will be called `theme` but for now (since not a built-in type) it would be `c_theme`.